test/integration/setup/adminclient/client.go

```diff
@@ -0,0 +1,305 @@
+CjAfBgNVHREEGDAWhhRzcGlmZmU6Ly9leGFtcGxlLm9yZzAKBggqhkjOPQQDBAOB
+jAAwgYgCQgHOtx4sNCioAQnpEx3J/A9M6Lutth/ND/h8D+7luqEkd4tMrBQgnMj4
+E0xLGUNtoFNRIrEUlgwksWvKZ3BksIIOMwJCAc8VPA/QYrlJDeQ58FKyQyrOIlPk
+Q0qBJEOkL6FrAngY5218TCNUS30YS5HjI2lfyyjB+cSVFXX8Szu019dDBMhV
+-----END CERTIFICATE-----
+`
+)
+
+var (
+	tdFlag         = flag.String("trustDomain", "domain.test", "server trust domain")
+	socketPathFlag = flag.String("socketPath", "unix:///tmp/agent.sock", "agent socket path")
+	serverAddrFlag = flag.String("serverAddr", "SPIRE-SERVER:8081", "server addr")
+	key            = testkey.MustEC256()
+)
+
+func main() {
+	flag.Parse()
+
+	// Create TD
+	td := spiffeid.RequireTrustDomainFromString(*tdFlag)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// Create X509Source
+	source, err := workloadapi.NewX509Source(ctx, workloadapi.WithClientOptions(workloadapi.WithAddr(*socketPathFlag)))
+	if err != nil {
+		log.Fatalf("Unable to create X509Source: %v", err)
+	}
+	defer source.Close()
+
+	// Create connection
+	tlsConfig := tlsconfig.MTLSClientConfig(source, source, tlsconfig.AuthorizeAny())
+	conn, err := grpc.DialContext(ctx, *serverAddrFlag, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
+	if err != nil {
+		log.Fatalf("Error creating dial: %v", err)
+	}
+	defer conn.Close()
+
+	// Run test for SVID RPCs
+	svidRPCs(ctx, conn, td)
+
+	// Run test for Bundle RPCs
+	bundleRPCs(ctx, conn, td)
+
+	// Run test for Entry RPCs
+	entryRPCs(ctx, conn, td)
+
+	// Run test for Agent RPCs
+	agentRPCs(ctx, conn, td)
+}
+
+// svidRPCs verify access to SVID RPCs
+func svidRPCs(ctx context.Context, conn *grpc.ClientConn, td spiffeid.TrustDomain) {
+	log.Println("Verifing SVID RPCs")
+	// Create SVID client
+	client := svid.NewSVIDClient(conn)
+
+	template := &x509.CertificateRequest{URIs: []*url.URL{td.NewID("new_workload").URL()}}
+	csr, err := x509.CreateCertificateRequest(rand.Reader, template, key)
+	if err != nil {
+		log.Fatalf("Fails to create CSR: %v", err)
+	}
+
+	_, err = client.MintX509SVID(ctx, &svid.MintX509SVIDRequest{
+		Csr: csr,
+	})
+	if err != nil {
+		log.Fatalf("Fails to mint X509 SVID: %v", err)
+	}
+
+	_, err = client.MintJWTSVID(ctx, &svid.MintJWTSVIDRequest{
+		Id:       &types.SPIFFEID{TrustDomain: td.String(), Path: "new_workload"},
+		Audience: []string{"myAud"},
+	})
+	if err != nil {
+		log.Fatalf("Fails to mint JWT SVID: %v", err)
+	}
+}
+
+// bundleRPCs verify access to Bundle RPCs
+func bundleRPCs(ctx context.Context, conn *grpc.ClientConn, td spiffeid.TrustDomain) {
+	log.Println("Verifing Bundle RPCs")
+	client := bundle.NewBundleClient(conn)
+
+	blk, _ := pem.Decode([]byte(testBundle))
+	if _, err := client.AppendBundle(ctx, &bundle.AppendBundleRequest{
+		Bundle: &types.Bundle{
+			TrustDomain: td.String(),
+			X509Authorities: []*types.X509Certificate{
+				{
+					Asn1: blk.Bytes,
+				},
+			},
+		},
+	}); err != nil {
+		log.Fatalf("Fails to append bundle: %v", err)
```

Comment:
I would change this and all the messages that start with "Fails to" to "Failed to"

```suggestion
		log.Fatalf("Failed to append bundle: %v", err)
```